tests/lib/mockMagic.py

```diff
@@ -9,3 +11,40 @@ def validate_config_for_device(self, device_mac, config):
             return return_validate_config_for_device['return_validate_intf']
         if config == validate_true['config']:
             return return_validate_config_for_device['return_validate_true']
+
+    def get_image_bundle_by_name(self, name):
+        """
+        mock to get image_bundle
+        """
+        if device_data[0]['imageBundle'] != 'Invalid_bundle_name':
+            return image_bundle
+        else:
+            return None
```

Comment:
so here I am confused for multiple things - it looks like device_data comes out of thin air - I know it comes from the data file but then I don't understand how it can be made "Invalid_bundle_name"

I don't see why this meythod has to use device_data at all.

For instance you could rather have


```suggestion
    def get_image_bundle_by_name(self, name):
        """
        mock to get image_bundle
        """
        if name != 'Invalid_bundle_name':
            return image_bundle
        else:
            return None
```

Which would make more sense and does not require to get any data from device_data which would look cleaner

Reply:
not sure wht this has been resolved?

Reply:
so, here I changed it to "name" as per your suggestion as it will have the same values as device_data[0]['imageBundle'] and to return None I am giving the string "Invalid_bundle_name" in input i.e. it is checking for the image which does not exist.